src/Microsoft.VisualStudio.Editor.Razor/TextBufferFactoryService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal abstract class TextBufferFactoryService
+    {
+        public abstract bool TryGetParser(ITextBuffer textBuffer, out VisualStudioRazorParser parser);
+
+        public abstract bool TryGetDocumentTracker(ITextBuffer textBuffer, out VisualStudioDocumentTracker documentTracker);
+
+        public abstract bool TryGetSmartIndenter(ITextBuffer textBuffer, out BraceSmartIndenter braceSmartIndenter);
+    }
```

Comment:
Let's call this `RazorEditorFactoryService` and make it public with just the `TryGetDocumentTracker` public